flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -180,10 +180,22 @@ public void run() {
 
 									// We overwrite existing files since this may be another attempt at fetching this archive.
 									// Existing files may be incomplete/corrupt.
+
+									//this is to save a job overview information, we put it in the ${path}/jobId folder
+									//because if put it in ${path}, new job inforamtion will override it and the web
+									//will show only one job all the same.
+									if (!target.toURI().toURL().getPath().contains(jobID)) {
```

Comment:
this isn't the right fix. This case _was_ handled at Line 165, but was rendered ineffective since the URL was changed.